src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java

```diff
@@ -320,6 +300,8 @@ public final XSDSchema buildSchemaInternal(
                                     schemaURIs.get(namespace),
                                     imports,
                                     includes);
+                            // add prefix
+                            schema.getQNamePrefixToNamespaceMap().put(prefix, namespace);
```

Comment:
Is this still needed ? If this is a complex feature type all the name-spaces should already have been added in the ``if`` above right ?

Reply:
Yes, it sounds sensible, deleting it.